cmd/buildkitd/main_unix.go

```diff
@@ -48,7 +54,13 @@ func listenFD(addr string, tlsConfig *tls.Config) (net.Listener, error) {
 }
 
 func traceSocketPath(root string) string {
-	return filepath.Join(root, "otel-grpc.sock")
+	p := filepath.Join(root, "otel-grpc.sock")
+	if len(p) > unixPathMax {
+		pp := filepath.Join(runtimeDir(), "buildkit-otel-grpc.sock")
+		logrus.Warnf("%q: unix socket path too long (> 104): fallback to %q", p, pp)
+		return pp
+	}
```

Comment:
Can we just deterministically use runtimeDir?

Symlink can be created to retain compatibility.